obspy/io/sac/sactrace.py

```diff
@@ -1105,12 +1102,15 @@ def read(cls, source, headonly=False, ascii=False, byteorder=None,
 
         .. rubric:: Example
 
-        >>> sac = SACTrace.read(filename, headonly=True) # doctest: +SKIP
-        >>> try: # doctest: +SKIP
-                sac.validate('data_hdrs') # doctest: +SKIP
-            except SacInvalidContentError: # doctest: +SKIP
-                sac = SACTrace.read(filename, headonly=False) # doctest: +SKIP
-                sac.validate('data_hdrs') # doctest: +SKIP
+        >>> from obspy.core.util import get_example_file
+        >>> from obspy.io.sac.util import SacInvalidContentError
+        >>> file_ = get_example_file("test.sac")
+        >>> sac = SACTrace.read(file_, headonly=True)
+        >>> try:
+        ...     sac.validate('data_hdrs')
+        ... except (ValueError, SacInvalidContentError):
+        ...     sac = SACTrace.read(file_, headonly=False)
+        ...     sac.validate('data_hdrs')
```

Comment:
Please take a look at this one. It seems you're expecting a `SacInvalidContentError`, but before this one gets raised it seems there's an (unexpected?) `ValueError` popping up. So it seems to me that in the underlying code you might want to catch that `ValueError` and raise a `SacInvalidContentError` instead? Don't know the code and expected bahavior enough to tell, though..